Dockerfile-go-deps

```diff
@@ -17,7 +17,8 @@ RUN GOPATH=${TEMP_GOPATH} bin/dep version
 # every dependency in `${TEMP_GOPATH}/pkg/dep` so it is important to remove it.
 # `go install` below cannot find the packages under vendor/ so move them to
 # /go/pkg. This is all done in a single RUN to avoid creating a giant
-# intermediate layer.
+# intermediate layer. `dep status` cannot be used here to ensure consistency
+# because it needs all our source files, which are not available here.
 COPY Gopkg.toml Gopkg.lock ./
 RUN \
   GOPATH=${TEMP_GOPATH} bin/dep ensure -vendor-only -v && \
```

Comment:
@siggy In the code you commented on I added a comment about why we can't use `dep status` here in Dockerfile-go-deps (AFAICT). I tried it and it always fails, probably because it requires all the source code for the current project in order to work.